DictDataGenerator.py

```diff
@@ -0,0 +1,92 @@
+import random
+import datetime
+import argparse
+import csv
+import os
+from DataGenerator import DataGenerator
+from DictDataClasses import DictDataClasses
+
+class DictDataGenerator(DataGenerator):
+
+    def __init__(self):
+        # Generate random inst ids with the prefix of the bank (ABC) and a random string composed of numbers and letters
+        self.dict = DictDataClasses().get_dict()
+
+    # In DataConfiguration.py, 'Data Args' field should look like:
+    # {'Type': 'position'}
+    def run(self, args):
+        type = args["Type"]
+        if type == 'price':
+            return self.__generate_price_entity()
+        elif type == 'position':
+            return self.__generate_position_entity()
+        elif type == 'inst-ref':
+            return self.__generate_inst_ref_entity()
+
+    @staticmethod
+    def main():
+        args = DictDataGenerator.__get_args()
+        dict_data_generator = DictDataGenerator()
+        dict_data_generator .__generate_data_files(args)
+
+    def __generate_data_files(self, args):
+        # Create out directory if it does not yet exist
+        if not os.path.exists('out'):
+            os.makedirs('out')
+
+        if args.prices > 0:
+            self.__create_data_file('out/prices.csv', args.prices, self.__generate_price_entity)
+        if args.positions > 0:
+            self.__create_data_file('out/positions.csv', args.positions, self.__generate_position_entity)
+        if args.inst_refs > 0:
+            self.__create_data_file('out/inst-ref.csv', args.inst_refs, self.__generate_inst_ref_entity)
+
+    @staticmethod
+    def __get_args():
+        parser = argparse.ArgumentParser()
+        parser.add_argument('--prices', nargs='?', type=int, default=0)
```

Comment:
For all of these please add a help parameter to help users understand what needs to be input.